meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor_git.bb

```diff
@@ -17,16 +19,28 @@ def mender_monitor_srcrev_from_src_uri(d, src_uri):
         src_uri_glob = src_uri_list[0][len("file://"):]
         # Get the filename
         filenames = glob.glob(src_uri_glob)
-        if len(filenames) != 1:
+        if len(filenames) == 0:
+            bb.error("Failed to find mender monitor on path %s" % src_uri_glob)
+            bb.error("Please make sure SRC_URI_pn-mender-monitor is pointing to the downloaded tarball ")
+        elif len(filenames) != 1:
             bb.error("Expected exactly one file, found: %s" % filenames)
-        filename = filenames[0]
-        # Now extract the git sha from the filename
-        m = re.match(r".*/mender-monitor-([0-9a-f]+)\.tar\.gz", filename)
-        if m is None:
-            # No match probably means that the tarball is a tagged version, in
-            # which case this recipe is not to be used but still needs to parse
+        filename = os.path.basename(filenames[0])
+        # Now extract the version from the filename
+        if filename == "mender-monitor-master.tar.gz":
+            # Building from external tarball, do not append git info
+            return ""
+        m = re.match(r"mender-monitor-([0-9]+\.[0-9]+\.[0-9]+(?:-build[0-9]+)?)\.tar\.gz", filename)
+        if m is not None:
+            # The tarball is a tagged version, in which case SRCREV is not to
+            # be used but still needs to parse
             return ""
-        return m.group(1)
+        m = re.match(r"mender-monitor-([0-9a-f]{7})\.tar\.gz", filename)
```

Comment:
I think that Git uses more characters once the repository reaches a certain size (run `git rev-parse --short HEAD` in the gui repo for an example). Not completely sure if that makes its way here, but maybe it's better to use `{7,}`?

Reply:
I didn't know about this! Makes sense. I'll amend.